openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java

```diff
@@ -1539,6 +1530,45 @@ private void handleEnumWithDescriptions(
     }
   }
 
+  private void validateTableType(JsonNode fieldValue, String propertyConfig) {
+    TableTypeConfig tableTypeConfig =
+        JsonUtils.convertValue(JsonUtils.readTree(propertyConfig), TableTypeConfig.class);
+    Set<String> configColumns = tableTypeConfig.getColumns();
+
+    JsonNode columnsNode = fieldValue.get("columns");
+    if (columnsNode == null
+        || !columnsNode.isArray() && columnsNode.size() < tableTypeConfig.getMinColumns()) {
+      throw new IllegalArgumentException(
+          "Minimum " + tableTypeConfig.getMinColumns() + " column required");
+    }
+
+    Set<String> fieldColumns = new HashSet<>();
+    columnsNode.forEach(column -> fieldColumns.add(column.asText()));
+
+    fieldColumns.removeAll(configColumns);
+    if (!fieldColumns.isEmpty()) {
+      throw new IllegalArgumentException(
+          "Expected columns: " + configColumns + ", but found undefined columns: " + fieldColumns);
+    }
+
+    JsonNode rowsNode = fieldValue.get("rows");
+    if (rowsNode == null || !rowsNode.isArray() && rowsNode.size() < tableTypeConfig.getMinRows()) {
```

Comment:
we can use the default value for rows from the jsonSchema right?

Reply:
yes @harshach the default values are coming from the json itself as declared in the config json like-

<img width="241" alt="image" src="https://github.com/user-attachments/assets/0259729e-bc6e-4dc6-b9b0-0c517da48756">